pandas/core/arrays/base.py

```diff
@@ -254,8 +256,9 @@ def _from_factorized(cls, values, original):
     # Must be a Sequence
     # ------------------------------------------------------------------------
 
-    def __getitem__(self, item):
-        # type (Any) -> Any
+    def __getitem__(
+        self, item: Union[int, slice, np.ndarray]
+    ) -> Union[ExtensionArray, Any]:
```

Comment:
should ExtensionArray be a typevar here?

Reply:
yes it should. Until there is consensus on usage have only added where needed to keep mypy green. see https://github.com/pandas-dev/pandas/pull/37898#discussion_r524486081, https://github.com/pandas-dev/pandas/pull/37817#discussion_r523401195, https://github.com/pandas-dev/pandas/pull/31728#discussion_r375587847, https://github.com/pandas-dev/pandas/pull/31384#discussion_r371964423 and others